array/array.go

```diff
@@ -223,56 +223,146 @@ func (a *String) ValueLen(i int) int {
 }
 
 type StringBuilder struct {
-	b *array.BinaryBuilder
+	b            *array.BinaryBuilder
```

Comment:
(Not really part of this review, but a thing I'd like to bring up) This is a thing I see in go more than I ever did in even C: what is `b`? Why do we prefer the single letter variable name over, say, `builder`?

Reply:
I think that's just something Go does? I don't really have an objection to changing this to `builder`. I just don't like typing that much.